src/daft-core/src/array/ops/utf8.rs

```diff
@@ -198,6 +198,19 @@ impl Utf8Array {
         Ok(Utf8Array::from((self.name(), Box::new(arrow_result))))
     }
 
+    pub fn reverse(&self) -> DaftResult<Utf8Array> {
+        let self_arrow = self.as_arrow();
+        let arrow_result = self_arrow
+            .iter()
+            .map(|val| {
+                let v = val?;
+                Some(v.chars().rev().collect::<String>())
```

Comment:
I believe that `chars()` actually gives you an iterator of characters rather than unicode characters
https://doc.rust-lang.org/std/primitive.str.html#method.chars

This may cause a panic due to non-valid UTF-8.

I believe you may actually want to use https://doc.rust-lang.org/std/primitive.str.html#method.char_indices
which will give you segments of what characters make up the next unicode character.

A good test for this is reversing a string of emojis or chinese characters

Reply:
`chars()` actually supports unicode characters well. I've added a handful of tests. Let me know if you think there's something still missing!

Reply:
Oh I see, thanks for checking!